tools/run_tests/artifacts/build_artifact_python.sh

```diff
@@ -280,4 +281,14 @@ then
   ${SETARCH_CMD} "${PYTHON}" src/python/grpcio_admin/setup.py \
       sdist bdist_wheel
   cp -r src/python/grpcio_admin/dist/* "$ARTIFACT_DIR"
+
+  # Build grpcio_observability source distribution
+  # Skips MacOS since grpcio_observability does not support MacOS.
+  if [ "$GRPC_UNIVERSAL2_REPAIR" == "" ]; then
```

Comment:
I see what you're doing here, but the name of this env var is misleading. Can we change the name to `GRPC_BUILD_MAC` or something like that? You'll have to also change [the place where we supply the environment variable](https://github.com/grpc/grpc/blob/1751f1043ed00cb064907bf6943c7fcad72516af/tools/run_tests/artifacts/artifact_targets.py#L145).